src/registry/registry.ts

```diff
@@ -123,6 +142,48 @@ export async function getRegisteredFieldSubmissionHandler(type: string): Promise
   return handler;
 }
 
+export async function getRegisteredformSchemaTransformers(): Promise<FormSchemaTransformer[]> {
+  const transformers = [];
+
+  // Fetch from cache if available
+  const cachedTransformers = registryCache.formTransformers;
+  if (Object.keys(cachedTransformers).length) {
+    for (const key in cachedTransformers) {
+      if (Object.prototype.hasOwnProperty.call(cachedTransformers, key)) {
+        transformers.push(cachedTransformers[key]);
+      }
+    }
+    return transformers;
+  }
+
+  //Get custom transformers
+  const formTransformersFromStore = getFormsStore().formTransformers;
+  if (Array.isArray(formTransformersFromStore)) {
+    const customTransformers = await Promise.all(
+      formTransformersFromStore.map(async (transformer) => {
+        const transformerImport = await transformer.load?.();
+        return transformerImport?.default;
+      }),
+    );
+    transformers.push(...customTransformers.filter((transformer) => transformer !== undefined));
+  }
+
+  if (inbuiltFormTransformers.length) {
+    for (const inbuiltTransformer of inbuiltFormTransformers) {
+      transformers.push(inbuiltTransformer.component);
+    }
+  }
+
+  // Cache all fetched transformers
+  for (const transformer of transformers) {
+    registryCache.formTransformers[
+      inbuiltFormTransformers.find((inbuiltTransformer) => inbuiltTransformer.component === transformer).name
+    ] = transformer;
+  }
```

Comment:
I don't understand why this logic is this complicated. What makes transformers special from other registry components? As an example here is a similar implementation for field handlers:

```ts
export async function getRegisteredFieldSubmissionHandler(type: string): Promise<SubmissionHandler> {
  if (registryCache.fieldSubmissionHandlers[type]) {
    return registryCache.fieldSubmissionHandlers[type];
  }
  let handler = inbuiltFieldSubmissionHandlers.find((handler) => handler.type === type)?.component;
  // if undefined, try serching through the registered custom handlers
  if (!handler) {
    const handlerImport = await getFormsStore()
      .fieldSubmissionHandlers.find((handler) => handler.type === type)
      ?.load?.();
    handler = handlerImport?.default;
  }
  registryCache.fieldSubmissionHandlers[type] = handler;
  return handler;
}
```

Reply:
From your guidance earlier, we had to get all the transformers at once.
The difference with  `getRegisteredFieldSubmissionHandler` is that it only returns one handler at a time

Reply:
I could be wrong here but getting all transformers in one shot can be like "Hey registry store, give me all you got". Something like:

```ts

const cache = [];

async function getRegisteredTransformers() {
  if (cache.length) {
    return cache;
  }

  const customTransformers = await Promise.all(
    formTransformersFromStore.map(async (registration) => {
      try {
        const transformerImport = await registration.load?.();
        return transformerImport?.default;
      } catch (error) {
        console.error("Error loading custom transformer:", error);
        return null; // Handle loading error gracefully
      }
    })
  );
  const allTransformers = [...inbuiltFormTransformers.map((registration) => registration.component), ...customTransformers.filter(Boolean)]; // Filter out null values

  // Cache the transformers for future use
  cache.push(...allTransformers);

  return allTransformers;
}
```

Reply:
@kajambiya did you take action on the above?